pkg/plugins/golang/v2/scaffolds/internal/templates/api/types.go

```diff
@@ -80,7 +80,7 @@ type {{ .Resource.Kind }}Status struct {
 
 // +kubebuilder:object:root=true
 // +kubebuilder:subresource:status
```

Comment:
I am not sure if we should change the V2 template. IMO: we should only perform changes in v3 to avoid we introduce a bug or an diff behaviour on it.

Reply:
`Resource.Namespaced` no longer exists, it is now `Resource.API.Namespaced`. You can see that no `testdata/project-v2*` file was modified.

Reply:
See: https://github.com/kubernetes-sigs/kubebuilder/pull/1696

I kept the Namespace there with a // todo: for we remove it when v2 is no longer supported

**Why?**

it reduces the other ifs and workarounds in the code to keep the same behaviour for v2 as the chance of we introduce a bug or a change that we should not for v2.

Reply:
There are several reasons what that approach doesn't fit this PR:
- There is no current defined on deprecating project configuratiom versions.
- Adding a duplicate field for this would require special cases in `Options.NewResource`, `Config.Marshal` for both v2 and v3, `Config.Unmarshal` for both v2 and v3, `Resource.Update`, and the added field to `Resource`.
- The only advantage that this has is not replacing `{{ .Resource.Namespaced }}` with `{{ .Resource.API.Namespaced }}` in `go/v2`.
- `go/v2` is **not** deprecated yet, so changes in the plugin should be perfectly fine while the scaffolded files are not modified, which is the case.